client/src/plugins/update-checks/UpdateChecks.js

```diff
@@ -0,0 +1,187 @@
+
+const UPDATE_CHECK_API = '/update-check';
+
+const HOURS_DENOMINATOR = 3600000;
+const HOURS_LIMIT = 24;
+
+export default class UpdateChecks extends PureComponent {
+
+  constructor(props) {
+    super(props);
+
+    if (!Flags.get(SERVER_INTERACTION) || !Flags.get(UPDATES_SERVER_URL)) {
+      return new NoopComponent();
+    }
+
+    this.state = {
+      showModal: false
+    };
+  }
+
+  async componentDidMount() {
+
+    const {
+      config
+    } = this.props;
+
+    const privacyPreferences = await config.get(PRIVACY_PREFERENCES_CONFIG_KEY);
+    if (!privacyPreferences || !privacyPreferences.ENABLE_UPDATE_CHECKS) {
+      this.setState({ checkNotAllowed: true });
+      return;
+    }
+
+    const latestUpdateCheckInfo = await config.get(LATEST_UPDATE_CHECK_INFO_CONFIG_KEY);
+    if (latestUpdateCheckInfo && !this.isTimeExceeded(latestUpdateCheckInfo.latestUpdateTime)) {
+      this.setState({ checkNotNeeded: true });
+      return;
+    }
+
+    this.setState({ isChecking: true });
+
+    const updatesServerUrl = Flags.get(UPDATES_SERVER_URL);
+    await this.checkLatestVersion(updatesServerUrl, latestUpdateCheckInfo);
+  }
+
+  async sendRequest(url) {
+    const response = await fetch(url, { method: 'GET' });
+    const responseJSON = await response.json();
+    return responseJSON;
+  }
+
+  async checkLatestVersion(updatesServerUrl, latestUpdateCheckInfo) {
+
+    const {
+      config,
+      _getGlobal
+    } = this.props;
+
+    try {
+      const editorID = await config.get(EDITOR_ID_CONFIG_KEY);
+      const modelerVersion = 'v' + Metadata.data.version;
+      const newerThan = latestUpdateCheckInfo ? (latestUpdateCheckInfo.latestCheckedVersion) : modelerVersion;
+      const osInfo = await config.get(OS_INFO_CONFIG_KEY);
+      const plugins = this.formatPlugins(_getGlobal('plugins').appPlugins);
+      const queryString = this.constructQueryString(editorID, modelerVersion, newerThan, osInfo, plugins);
+
+      const url = Flags.get(UPDATES_SERVER_URL) + UPDATE_CHECK_API + queryString;
+      const responseJSON = await this.sendRequest(url);
+
+      const latestVersion = responseJSON.latestVersion;
+      let newLatestUpdateCheckInfo = latestUpdateCheckInfo ? latestUpdateCheckInfo : {};
+      if (latestVersion) {
+        const downloadURL = responseJSON.downloadURL;
+        const releases = responseJSON.releases;
+        this.setState({
+          isChecking: false,
+          showModal: true,
+          latestVersionInfo: { latestVersion, downloadURL, releases },
+          currentVersion: modelerVersion
+        });
+        newLatestUpdateCheckInfo.latestCheckedVersion = latestVersion;
+      }
+
+      newLatestUpdateCheckInfo.latestUpdateTime = new Date().getTime();
+      config.set(LATEST_UPDATE_CHECK_INFO_CONFIG_KEY, newLatestUpdateCheckInfo);
+    } catch (err) {
+      this.setState({ isChecking: false, requestError: true });
+    }
+  }
+
+  appendBasicQueryParameter(queryParam, key, value) {
```

Comment:
How about if we use URL API? 

https://developer.mozilla.org/en-US/docs/Web/API/URL

Reply:
And I think the whole request logic could be exported to a separate component like we do in the deployment plugin. IMO that would make the code easier to consume and maintain.

Reply:
> How about if we use URL API?
> 
> https://developer.mozilla.org/en-US/docs/Web/API/URL

But how about IE support :P

Just kidding, nice suggestion.